apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -312,11 +312,39 @@ def bind_port(self, context):
                       vnic_type)
             return
 
-        # Attempt to bind ports for DVS agents for nova-compute daemons
-        # first. This allows having network agents (dhcp, metadata)
-        # that typically run on a network node using an OpFlex agent to
-        # co-exist with nova-compute daemons for ESX, which host DVS agents.
         if context.current['device_owner'].startswith('compute:'):
+            # enforce the secure port binding rules if possible
+            if context.current['device_id']:
+                vm = nova_client.NovaClient().get_server(
+                    context.current['device_id'])
+                network_id = context.current.get('network_id')
+                net = self._get_plugin().get_network(context._plugin_context,
+                                                     network_id)
+                vrf = self._get_network_vrf(context, net)
+                aci_tenant = self.apic_manager.apic.fvTenant.name(
+                    vrf['aci_tenant'])
+                aci_vrf = self.apic_manager.apic.fvCtx.name(vrf['aci_name'])
+
+                ok_to_bind = True
+                secure_pb_list = self.net_cons.get_allow_vm_names(
```

Comment:
You seem to be querying the allowed VMs using the mapped ACI tenant etc., however the constraints file is organized in terms of OpenStack tenants and networks because those are the terms that an Openstack user is familiar with. It also avoids the need for Openstack user to know how exactly the Openstack resource will get mapped in ACI (think about the different modes and scoping options). Can you not use OS tenants and networks?